src/briefcase/platforms/iOS/xcode.py

```diff
@@ -512,28 +511,34 @@ def run_app(
         # Try to uninstall the app first. If the app hasn't been installed
         # before, this will still succeed.
         self.logger.info(f"Installing {label}...", prefix=app.app_name)
-        try:
-            with self.input.wait_bar("Uninstalling any existing app version..."):
-                self.tools.subprocess.run(
-                    ["xcrun", "simctl", "uninstall", udid, app.bundle_identifier],
-                    check=True,
+        with self.input.wait_bar(
+            "Uninstalling any existing app version..."
+        ) as keep_alive:
+            uninstall_popen = self.tools.subprocess.Popen(
+                ["xcrun", "simctl", "uninstall", udid, app.bundle_identifier]
+            )
+            with uninstall_popen:
```

Comment:
Any reason to no use a nested or multi-clause with statement here? e.g.,
```
with self.input.wait_bar(
             "Uninstalling any existing app version..."
         ) as keep_alive, self.tools.subprocess.Popen(
             ["xcrun", "simctl", "uninstall", udid, app.bundle_identifier]
         ) as uninstall_popen:
```

Reply:
Because it's harder to mock... 🥲 

But let me take a look at exactly what'll it take...should just be a few copy/pastes once I get it

Reply:
Re-implemented using multi-with blocks.